akvo/rsr/spa/app/modules/editor/section1/comp/project-picker.jsx

```diff
@@ -41,72 +60,66 @@ const ProjectPicker = ({ loading, projects, savedData, removeSetItem, hasImporte
       }, 300)
     }
   }
-  const removeItem = (input, fields) => {
-    removeSetItem(1, 'relatedProjects', 0)
-    fields.remove(0)
-  }
-  return [
-    <FieldArray name="relatedProjects">
-      {({ fields }) => (
+  return (
+    <Skeleton loading={!(projects)} paragraph={{ rows: 2 }} active>
       <Item label={(<InputLabel optional>{t('Contributes to')}</InputLabel>)}>
-        <AutoSave sectionIndex={1} setName="relatedProjects" itemIndex={0} />
+        <AutoSave sectionIndex={1} />
         {isExternal && (
-          <FinalField
-            placeholder={t('IATI Identifier')}
-            name="relatedProjects[0].relatedIatiId"
-            control="input"
+          <Field
+            name="externalParentIatiActivityId"
+            render={({ input }) => (
+              <FinalField
+                {...input}
+                placeholder={t('IATI Identifier')}
+                name="externalParentIatiActivityId"
+                control="input"
+              />
+            )}
           />
         )}
         {!isExternal && (
           <Field
-            name="relatedProjects[0].relatedProjectName"
-            render={(nameProps) => {
+            name="contributesToProject"
+            render={({ input }) => {
+              const $options =
+                projects && projects.length > 0
+                  ? ((input.value && state.searchStr.length === 0) ? [projects.find(it => it.id === input.value)].filter(it => it != null).map(({ id, title }) => ({ value: id, label: `[${id}] ${title ? title : t('Untitled project')}` })) : state.options)
```

Comment:
I know this line probably isn't yours, but breaking it up would probably help, because double ternary statements are very difficult to read and it's way too long.

Reply:
All right, working on it